components/dashboard/dashboard.go

```diff
@@ -281,3 +287,30 @@ func (d *Dashboard) cleanOauthClient(cli client.Client, dscispec *dsciv1.DSCInit
 	}
 	return nil
 }
+
+func (d *Dashboard) IsInstalled(owner metav1.Object) (bool, error) {
```

Comment:
I can be totally wrong, but at the first glance it looks like other way around.
I mean, DataScienceCluster object way down inside operator code serves as owner only in deploy in SetControllerReference(), so ReconcileComponent() and below can take it, then the method will be DataScienceCluster.IsInstalled(componentName string) or (component ComponentInterface)

Reply:
It can't take `DataScienceCluster` type because of the package import cycle limitation of `golang`. I assume that is why it's defined as super type `metav1.Object` as it's not documented why the `owner` is needed.

Lets break it down. Original contract was:

`ReconcileComponent(ctx context.Context, cli client.Client, resConf *rest.Config, owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec, currentComponentStatus bool)`

This is called from the reconcile as follows:

https://github.com/opendatahub-io/opendatahub-operator/blob/c04130fbf618d0cf71ad076904eb869802315a60/controllers/datasciencecluster/datasciencecluster_controller.go#L281

We are passing `instance` (of DSC) as an owner (and that is the only owner used AFAIK) and then we pass the value of `instance.Status.InstalledComponents[componentName]`

My line of thinking here is that instead of polluting the API with that last boolean flag we could move this logic down to the only component which is now in need of this information. And in that method simply do the `instance.Status.InstalledComponents[componentName]` lookup.

Reply:
Though maybe `func (d *Dashboard) IsInstalled(owner metav1.Object) (bool, error)` naming is confusing here. I will think about it.

Reply:
@ykaliuta does it clear the confusion or am I still on a different train of thought? :)

Reply:
I would not say I was confused, just had a bit different view on design, so go ahead :)

Reply:
i am not sure i understand this confusion. 
is the confusion coming from the name of the method? or why we should have this method ? or a method to DSC instead of a generic function?

Reply:
> i am not sure i understand this confusion. is the confusion coming from the name of the method? or why we should have this method ? or a method to DSC instead of a generic function?

A method to component instead of a method to DSC.

My way of thinking was: if component stored its own status or reference to the owner, than asking the status with "if I am installed()" sounds fine. But asking "if I am installed to (some owner)" is a bit clumsy. While asking DSC then "do you have installed(component)" more natural.

But the proposal fine for me, no problem (and as described other way won't work).